### PR TITLE
Ported the plugin youtube.py from branch master

### PR DESCRIPTION
**Pull request by HitaloSama**
This plugin has not been tested on EduuRobot, I did tests on my bot (PyKorone) only if there is any compatibility error with the base of EduuRobot let me know.

### CODE REVIEW

**Review by alissonlauffer — CHANGES_REQUESTED** [review]
Overall it looks good, but there are some issues I can see for now:
 - youtube-dl is missing in requirements.txt file.
 - You've added ytdl to commands list but forgot to create a description for it.
 - youtube-dl is not running in an executor and will block the pool (see userlixo-rfc ytdl.py).

About the second one:
You must create a new json file in `locales/en-GB/youtube.py` with:

```json
{
    "ytdl_description": "command description here"
}
```

File: plugins/youtube.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            )
+            ydl.download(["https://www.youtube.com/watch?v=" + yt["id"]])
+            await sent_id.edit_text("Sending audio...")
+            await c.send_chat_action(chat_id=m.chat.id, action="upload_video")
+            await c.send_audio(
+                chat_id=m.chat.id,
+                audio=open(ydl.prepare_filename(yt), "rb"),
+                performer=performer,
+                title=title,
+                duration=yt["duration"],
```
Comment (alissonlauffer) [inline plugins/youtube.py:53]:
```suggestion
                duration=int(yt["duration"]),
```

Pyrogram uses ints for duration, whereas youtube-dl uses floats. This resolves the `AttributeError: 'float' object has no attribute 'to_bytes'` Exception

**Comment by alissonlauffer** [comment]
Any update on this? Thanks.

**Comment by HitaloSama** [comment]
@alissonlauffer i tested what you told me for youtube-dl to be async but it didn't help, easily someone could kill the bot or slow down.

**Pull request closed without merging**